airbyte-integrations/connectors/source-confluence/source_confluence/manifest.yaml

```diff
@@ -157,3 +199,15 @@ spec:
         description: "Your Confluence domain name"
         type: string
         order: 2
+      space_name:
+        type: string
+        title: "Space Name"
+        description: 'The name of the Space you wish to extract.'
+        examples: ["Sample_Space"]
+        order: 3
+      start_date:
+        type: string
+        pattern: ^[0-9]{4}-[0-9]{2}-[0-9]{2} [0-9]{2}:[0-9]{2}$
+        description: 'UTC date and time in the format "yyyy-mm-dd HH:MM". Any data before this date will not be replicated.'
+        examples: ["1970-01-01 01:15"]
+        order: 4
```

Comment:
Both parameters are required by the `search` stream. @flash1293 what are you opinion here?

Reply:
Looks like we can make the space actually optional by not adding them to the cql query if they are not set using the `~` operator to concatenate streams:
```
cql: type=page {{  ("and space="  ~ config['space_name']) if config['space_name'] }} {{ ('and lastmodified >= "' ~ (stream_state.lastUpdated or config['start_date']) ~ '"') if stream_state.lastUpdated or config['start_date'] }}
```

The start date is required by the incremental sync module, but we could just set some default that will essentially be "all the data" like `2000-01-01 00:00`. This way we don't cause a breaking change. What do you think?

edfincham

Reply:
That looks great! Good to see some more advanced usage of the string formatting too - worth adding to the docs :slightly_smiling_face: